pkg/kubernetes/client/nodes.go

```diff
@@ -0,0 +1,164 @@
+package client
+
+import (
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/pod-security-admission/api"
+)
+
+func (client *OVNKMCPServerClientSet) DebugNode(ctx context.Context, name, image string, command []string, isOpenshift bool) (string, string, error) {
+	namespace := metav1.NamespaceDefault
+	cleanupNamespace := func() {}
+	var err error
+	if isOpenshift {
+		namespace, cleanupNamespace, err = client.createNamespace(ctx)
+		if err != nil {
+			return "", "", err
+		}
+	}
+	debugPodName, cleanupPod, err := client.createPod(ctx, name, namespace, image, isOpenshift)
+	if err != nil {
+		return "", "", err
+	}
+
+	// Execute the command in the pod.
+	stdout, stderr, err := client.ExecPod(ctx, debugPodName, namespace, "debug-container", command)
+	if err != nil {
+		return "", "", err
+	}
+
+	if cleanupPod != nil {
+		cleanupPod()
+	}
+
+	if cleanupNamespace != nil {
+		cleanupNamespace()
+	}
+
+	return stdout, stderr, nil
+}
+
+func (client *OVNKMCPServerClientSet) createNamespace(ctx context.Context) (string, func(), error) {
+	tmpNS := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "openshift-debug-",
+			Labels: map[string]string{
+				api.EnforceLevelLabel:                            string(api.LevelPrivileged),
+				api.AuditLevelLabel:                              string(api.LevelPrivileged),
+				api.WarnLevelLabel:                               string(api.LevelPrivileged),
+				"security.openshift.io/scc.podSecurityLabelSync": "false",
+			},
+			Annotations: map[string]string{
+				"openshift.io/node-selector": "",
+			},
+		},
+	}
+	ns, err := client.clientSet.CoreV1().Namespaces().Create(ctx, tmpNS, metav1.CreateOptions{})
+	if err != nil {
+		return "", nil, err
+	}
+
+	cleanupNamespace := func() {
+		// Delete the namespace.
+		client.clientSet.CoreV1().Namespaces().Delete(ctx, ns.Name, metav1.DeleteOptions{})
```

Comment:
I think we should at least log the returned error as warning instead of silencing it.